gobblin-runtime/src/main/java/org/apache/gobblin/runtime/DatasetTaskSummary.java

```diff
@@ -17,30 +17,26 @@
 
 package org.apache.gobblin.runtime;
 
+import lombok.Data;
+
+import org.apache.gobblin.metrics.DatasetMetric;
+
+
 /**
  * A class returned by {@link org.apache.gobblin.runtime.SafeDatasetCommit} to provide metrics for the dataset
  * that can be reported as a single event in the commit phase.
  */
+@Data
 public class DatasetTaskSummary {
   private final String datasetUrn;
   private final long recordsWritten;
   private final long bytesWritten;
+  private final boolean datasetCommitSucceeded;
 
-  public DatasetTaskSummary(String datasetUrn, long recordsWritten, long bytesWritten) {
-    this.datasetUrn = datasetUrn;
-    this.recordsWritten = recordsWritten;
-    this.bytesWritten = bytesWritten;
-  }
-
-  public String getDatasetUrn() {
-    return datasetUrn;
-  }
-
-  public long getRecordsWritten() {
-    return recordsWritten;
-  }
-
-  public long getBytesWritten() {
-    return bytesWritten;
+  /**
+   * Convert a {@link DatasetTaskSummary} to a {@link DatasetMetric}.
+   */
+  public static DatasetMetric toDatasetMetric(DatasetTaskSummary datasetTaskSummary) {
```

Comment:
NBD, but why not an instance method?  (invocation syntax, inside `.map()` would remain unchanged)

Reply:
if it's an instance method the syntax inside map would be different since it wouldn't be static anymore right? It should look like `.map(summary -> summary.toDatasetMetric())`

Reply:
syntax should be no different - https://docs.oracle.com/javase/tutorial/java/javaOO/methodreferences.html
e.g.
```
String::compareToIgnoreCase
String::concat
```